scripts/test/ISISPowderRunDetailsTest.py

```diff
@@ -0,0 +1,127 @@
+
+        yaml_callable = RunDetails.RunDetailsFuncWrapper(function=wrapped_funcs.get_cal_mapping_dict,
+                                                         run_number_string=run_number_string, inst_settings=mock_inst)
+
+        empty_callable = yaml_callable.add_to_func_chain(function=wrapped_funcs.cal_dictionary_key_helper,
+                                                         key="custom_empty_run_numbers")
+        vanadium_callable = yaml_callable.add_to_func_chain(function=wrapped_funcs.cal_dictionary_key_helper,
+                                                            key="custom_vanadium_run_numbers")
+        grouping_callable = RunDetails.RunDetailsFuncWrapper(function=lambda: expected_grouping_file_name)
+
+        output_obj = RunDetails.create_run_details_object(run_number_string=run_number_string, inst_settings=mock_inst,
+                                                          is_vanadium_run=True, empty_run_call=empty_callable,
+                                                          vanadium_run_call=vanadium_callable,
+                                                          grouping_file_name_call=grouping_callable)
+
+        self.assertEqual(output_obj.label, expected_label)
+        self.assertEqual(output_obj.empty_runs, expected_empty_runs)
+        self.assertEqual(output_obj.grouping_file_path,
+                         os.path.join(mock_inst.calibration_dir, expected_grouping_file_name))
+        self.assertEqual(output_obj.vanadium_run_numbers, expected_vanadium_runs)
+        self.assertEqual(output_obj.run_number, expected_vanadium_runs)
+
+    def test_run_details_splined_name_list_is_used(self):
+        expected_vanadium_runs = "11-12"
+        expected_offset_file_name = "offset_file_name"
+        splined_name_list = ["bar", "bang", "baz"]
+        mock_inst = self.setup_mock_inst_settings(yaml_file_path="ISISPowderRunDetailsTest.yaml")
+        output_obj = RunDetails.create_run_details_object(run_number_string=10, inst_settings=mock_inst,
+                                                          is_vanadium_run=False, splined_name_list=splined_name_list)
+
+        expected_splined_out_str = ''.join('_' + val for val in splined_name_list)
+        expected_output_name = "VanSplined_" + expected_vanadium_runs + expected_splined_out_str
+        expected_output_name += '_' + expected_offset_file_name + ".nxs"
+        expected_path = os.path.join(mock_inst.calibration_dir, output_obj.label, expected_output_name)
+        self.assertEqual(expected_path, output_obj.splined_vanadium_file_path)
+
+
+class MockInstSettings(object):
+    def __init__(self, cal_file_path, calibration_dir):
+        self.calibration_dir = calibration_dir
+        self.cal_mapping_path = cal_file_path
+        self.grouping_file_name = MockInstSettings.gen_random_string()
+
+    @staticmethod
+    def gen_random_string():
+        return ''.join(random.choice(string.ascii_lowercase) for _ in range(10))
```

Comment:
Can save another 0.5s per call here by making the code inside `join(...)` a list:

```
In [66]: timeit("''.join(random.choice(string.ascii_lowercase) for _ in range(10))",setup='import random, string; splined_name_list = ["name","2","fff"]')
Out[66]: 6.115681886672974
```
```
In [67]: timeit("''.join([random.choice(string.ascii_lowercase) for _ in range(10)])",setup='import random, string; splined_name_list = ["name","2","fff"]')
Out[67]: 5.514591932296753
```